website/client/src/components/challenges/challengeModal.vue

```diff
@@ -556,14 +557,21 @@ export default {
       challengeDetails.categories = serverCategories;
 
       let challenge;
-      if (this.cloning) {
-        challenge = await this.$store.dispatch('challenges:cloneChallenge', {
-          challenge: challengeDetails,
-          cloningChallengeId: this.cloningChallengeId,
-        });
-        this.cloningChallengeId = '';
-      } else {
-        challenge = await this.$store.dispatch('challenges:createChallenge', { challenge: challengeDetails });
+      try {
+        if (this.cloning) {
+          challenge = await this.$store.dispatch('challenges:cloneChallenge', {
+            challenge: challengeDetails,
+            cloningChallengeId: this.cloningChallengeId,
+          });
+          this.cloningChallengeId = '';
+        } else {
+          challenge = await this.$store.dispatch('challenges:createChallenge', { challenge: challengeDetails });
+        }
+      } catch (e) {
```

Comment:
thanks @aristizabal95 ! I see that you're returning inside the `catch` block which makes sense but it'll prevent these 2 lines from being executed while they probably should https://github.com/HabitRPG/habitica/pull/11769/files#diff-a73635db46b2d5f7643520a5e8ebcbf1R601-R603


It might be a good idea to refactor this to use a try/catch/finally flow with all the cleanup that is shared by the success and failure state in the `finally` block

Reply:
Hi @plagias, sure! Could do that change, although I would like to discuss it from an UX perspective. 

At the moment, if a server-side error is thrown during the challenge creation an alert is displayed describing such error and the form stays up for the user to correct the changes. Adding the `finally` block clearing the working challenge contents and closing the form modal would remove all unsaved progress, forcing the user to rewrite everything from scratch if they made a mistake.

Now, with the changes I made an user could just resubmit the form without any change, which may also not be good UX. I'd say a good solution would be to keep the submit button disabled until a change is made on any of the inputs (preferably on the input that's causing errors, but I would need to know which one was from the server). I believe this was the preferred behaviour for issue #11545 .

Tell me your opinion and I'll get working on your preferred solution!

Reply:
@aristizabal95 we defenitely want people to be able to correct the error and resubmit, I missed that part :)

> I'd say a good solution would be to keep the submit button disabled until a change is made on any of the inputs (preferably on the input that's causing errors, but I would need to know which one was from the server)

This looks like a good idea, I'd say for now we're not going to change the server response so let's stick to keeping it disabled until at least an input is modified

thanks!